server/routes/update-goal/UpdateGoalController.ts

```diff
@@ -57,10 +57,12 @@ export default class UpdateGoalController {
 
     const goalData: Partial<NewGoal> = {
       steps: updatedSteps,
-      note,
+      note: note || null,
```

Comment:
this works but with the backend changes should not be necessary any more. OK to keep it if you prefer!